### PR TITLE
folder restructuring

### PR DESCRIPTION
**Pull request by tjhei**
- move cmake related scripts to ./cmake/ and ./tests/cmake/
- add some documentation

### CODE REVIEW

**Comment by gassmoeller** [comment]
I am not a cmake expert, but if it works for you and the tester is ready this is ready to go, I guess.
Will build directories figure out the change, or does everybody has to delete the content of the build folder and rerun cmake?

**Comment by tjhei** [comment]
Rerunning cmake will copy the files and it should just work. (famous last words)

File: tests/cmake/default.sh [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
+
+# This is the default processing script, which just prints the output unmodified.
```
Comment (bangerth) [inline tests/cmake/default.sh]: processing *what*?

**Comment by bangerth** [comment]
OK with the one addition to the comment.

File: tests/cmake/default.sh (hunk 1 above)
Comment (tjhei) [inline tests/cmake/default.sh]: clarified it.

**Comment by gassmoeller** [comment]
ok it works on my machine. Ready to merge

**Pull request merged**